autoarray/structures/arrays/kernel_2d.py

```diff
@@ -481,33 +485,90 @@ def convolved_array_from(self, array: Array2D) -> Array2D:
 
         return Array2D(values=convolved_array_1d, mask=array_2d.mask)
 
-    def convolved_array_with_mask_from(self, array: Array2D, mask: Mask2D) -> Array2D:
+    def convolve_image(self, image, blurring_image, jax_method="fft"):
         """
-        Convolve an array with this Kernel2D
+        For a given 1D array and blurring array, convolve the two using this psf.
 
         Parameters
         ----------
         image
-            An array representing the image the Kernel2D is convolved with.
+            1D array of the values which are to be blurred with the psf's PSF.
+        blurring_image
+            1D array of the blurring values which blur into the array after PSF convolution.
+        jax_method
+            If JAX is enabled this keyword will indicate what method is used for the PSF
+            convolution. Can be either `direct` to calculate it in real space or `fft`
+            to calculated it via a fast Fourier transform. `fft` is typically faster for
+            kernels that are more than about 5x5. Default is `fft`.
+        """
 
-        Returns
-        -------
-        convolved_image
-            An array representing the image after convolution.
+        slim_to_native = jnp.nonzero(
+            jnp.logical_not(image.mask.array), size=image.shape[0]
+        )
+        slim_to_native_blurring = jnp.nonzero(
+            jnp.logical_not(blurring_image.mask.array), size=blurring_image.shape[0]
+        )
 
-        Raises
-        ------
-        KernelException if either Kernel2D psf dimension is odd
+        expanded_array_native = jnp.zeros(image.mask.shape)
+
+        expanded_array_native = expanded_array_native.at[slim_to_native].set(
+            image.array
+        )
+        expanded_array_native = expanded_array_native.at[slim_to_native_blurring].set(
+            blurring_image.array
+        )

```

Comment:
Might be worth checking if `jnp.where` can ends up doing this faster after a `jit`?  Or if it is the same speed, you can reduce this to two lines of code (just double check the lines below to make sure they do the same thing).

```python
expanded_array_native = jnp.where(image.mask.array, 0, image.array)
expanded_array_native = jnp.where(blurring_image.mask.array, expanded_array_native, blurring_image.array)
```

Reply:
I just double checked this, and I think the index ordering when using `jnp.where` is not quite the same as what you have, so my suggested change might not work as I was expecting.

Reply:
And with more testing, I have found since we have a 2D mask and a 1D slim from the `jnp.where` does not work as intended.  Keep the code as you have it.

Reply:
Cool, we may be able to produce images in 2D, perform convolution and then flatten.

Storing everything in 1D throughout the whole likelhood function was optimal for `numba` because we could store in memory a lot of arrays that did the mappings for us. But I don't think this is very JAX-y.